presto-main/src/main/java/com/facebook/presto/memory/DefaultQueryContext.java

```diff
@@ -320,17 +326,42 @@ private boolean tryReserveMemoryNotSupported(String allocationTag, long bytes)
         throw new UnsupportedOperationException("tryReserveMemory is not supported");
     }
 
-    private static void enforceUserMemoryLimit(long allocated, long delta, long maxMemory)
+    private void enforceUserMemoryLimit(long allocated, long delta, long maxMemory)
     {
         if (allocated + delta > maxMemory) {
-            throw exceededLocalUserMemoryLimit(succinctBytes(maxMemory), succinctBytes(allocated), succinctBytes(delta));
+            throw exceededLocalUserMemoryLimit(succinctBytes(maxMemory), getAdditionalFailureInfo(allocated, delta));
         }
     }
 
-    private static void enforceTotalMemoryLimit(long allocated, long delta, long maxMemory)
+    private void enforceTotalMemoryLimit(long allocated, long delta, long maxMemory)
     {
         if (allocated + delta > maxMemory) {
-            throw exceededLocalTotalMemoryLimit(succinctBytes(maxMemory), succinctBytes(allocated), succinctBytes(delta));
+            throw exceededLocalTotalMemoryLimit(succinctBytes(maxMemory), getAdditionalFailureInfo(allocated, delta));
         }
     }
+
+    private synchronized Optional<String> getAdditionalFailureInfo(long allocated, long delta)
+    {
+        Map<String, Long> queryAllocations = memoryPool.getTaggedMemoryAllocations().get(queryId);
```

Comment:
is every operator allocation tagged today?
if not, the mssage should say "Top Tagged Consumers" instead of "Top Consumers"